### DISCUSSION TITLE
Network activechain is undefined on initial load

### DISCUSSION THREAD

**Discussion by kkagill** [wevm/wagmi#1354] [Q&A]
```
  const { chain: activeChain } = useNetwork();
  
  useEffect(() => {
    console.log(activeChain?.id)
  }, []);
```
I want to detect which network user is currently on when they visited the site.
But on initial load, it is undefined. Any idea?

**Comment by hankbae93** [comment]
did you check useAccount? not connect not network 
if connect account, you'll see network object
